### PR TITLE
trust-dns: rebrand as hickory-dns

### PR DESCRIPTION
**Pull request by oneingan**
## Description of changes

Trust DNS project has been recently rebranded to Hickory DNS. This [blog post](https://bluejekyll.github.io/blog/posts/announcing-hickory-dns/) explains the reasoning behind this move.

I have staged more changes to Hickory DNS package and NixOS Module but prefer for this PR only cover the rebranding stuff.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [X] `sandbox = true`
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [X] (Package updates) Added a release notes entry if the change is major or breaking
  - [X] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/rename.nix [hunk 1]
```diff
@@ -127,6 +127,7 @@ in
     (mkRemovedOptionModule [ "services" "tedicross" ] ''
       The corresponding package was broken and removed from nixpkgs.
     '')
+    (mkRemovedOptionModule [ "services" "trust-dns" ] "Package has been completely rebranded by upstream as hickory-dns, and thus the service and the package were renamed in NixOS as well.")
```
Comment (SuperSandro2000) [inline nixos/modules/rename.nix:130]: Can't we make this a mkRenamedOption?

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -874,6 +874,7 @@ mapAliases ({
   tokodon = plasma5Packages.tokodon;
   tor-browser-bundle-bin = tor-browser; # Added 2023-09-23
   transfig = fig2dev; # Added 2022-02-15
+  trust-dns = throw "trust-dns is rebranded as hickory-dns"; # Added 2023-10-20
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:877]:
```suggestion
  trust-dns = hickory-dns; # Added 2023-10-20
```
This should be a noop, right?

File: nixos/modules/rename.nix (hunk 1 above)
Comment (oneingan) [reply nixos/modules/rename.nix]: done

**Review by adamcstephens — APPROVED** [review]

File: nixos/modules/services/networking/hickory-dns.nix [hunk 3]
```diff
@@ -126,23 +135,23 @@ in
...
+      description = "hickory-dns Domain Name Server";
+      unitConfig.Documentation = "https://hickory-dns.org/";
       serviceConfig = {
         ExecStart =
         let
           flags =  (lib.optional cfg.debug "--debug") ++ (lib.optional cfg.quiet "--quiet");
           flagsStr = builtins.concatStringsSep " " flags;
         in ''
-          ${cfg.package}/bin/trust-dns --config ${configFile} ${flagsStr}
+          ${cfg.package}/bin/hickory-dns --config ${configFile} ${flagsStr}
```
Comment (uninsane) [inline nixos/modules/services/networking/hickory-dns.nix:147]: without bumping the package version, this change doesn't work. 0.23.0 still ships only `bin/trust-dns` and not `bin/hickory-dns`. and 0.24.0 ships `bin/hickory-dns` only, without `bin/trust-dns`. so unless we want to introduce and leverage `meta.mainProgram`, there has to be a patch which simultaneously upgrades 0.23.0 -> 0.24.0 and changes the name here.

**Review by uninsane — CHANGES_REQUESTED** [review]

File: nixos/modules/services/networking/hickory-dns.nix (hunk 3 above)
Comment (uninsane) [reply nixos/modules/services/networking/hickory-dns.nix:147]: here's a PR which, if merged before the rebrand, would address this issue: <https://github.com/NixOS/nixpkgs/pull/263054>

Comment (uninsane) [reply nixos/modules/services/networking/hickory-dns.nix:147]: PR above has been merged, should be safe to just leave this line as it's set in master

**Comment by uninsane** [comment]
@oneingan if you get a chance to rebase this patch, it should be good to go. i can give it a deploy and then sign off.

**Comment by SuperSandro2000** [comment]
@oneingan can you please rebase the PR?

**Comment by oneingan** [comment]
Apologies for the wait. The situation is that initially, we intended to implement trust-dns in our company, but we opted for a different solution. If you're still comfortable with it, I can attempt to merge this, but please note that I may not be able to provide ongoing maintenance for it.

**Comment by uninsane** [comment]
> please note that I may not be able to provide ongoing maintenance for it.

i'm committed to maintaining trust-dns/hickory-dns for the indefinite future, so no problem there. if you're still using trust-dns in a more limited context or otherwise want to provide review/testing on future package updates, that's more than okay, otherwise just remove your name from `maintainers = [ colinsane juaningan ]` when doing the rebase and don't worry about it :)

**Comment by aktaboot** [comment]
would be nice to have this merged, if there isn't much blocking it

**Comment by uninsane** [comment]
rebased and resurrected as a [new PR](https://github.com/NixOS/nixpkgs/pull/316466) since i don't have permissions to update this one. i've kept @oneingan as the commit author.

**Pull request closed without merging**